### DISCUSSION TITLE
YAML-to-JSON and JSON-to-YAML conversion

### DISCUSSION THREAD

**Discussion by electriquo** [TomWright/dasel#238] [Q&A] [answered]
Some other tools, convert YAML-to-JSON and JSON-to-YAML and even XML-to-JSON/YAML

```shell
$ echo 'foo: bar' | yq -c
{"foo":"bar"}

$ echo 'foo: bar' | yq -c | yq -y
foo: bar
```

I would love to see `dasel` do the same and rely only on `dasel`.
What do you think?

**Comment by TomWright** [comment, marked as answer]
This is already possible using the `-r`,`--read` and `-w`,`--write` flags. You can read more on them here: https://daseldocs.tomwright.me/usage/select

```bash
$ echo 'foo: bar' | dasel -r yaml -w json -c
{"foo":"bar"}

$ echo 'foo: bar' | dasel -r yaml -w toml
foo = "bar"

$ echo 'foo: bar' | dasel -r yaml -w json | dasel -r json -w yaml
foo: bar

$ echo '<users><user><name>Tom</name></user><user><name>Jim</name></user></users>' | dasel -r xml -w yaml
users:
  user:
  - name: Tom
  - name: Jim
```

The supported file types can be found here: https://daseldocs.tomwright.me/usage/supported-file-types
